blpapijs.cpp

```diff
@@ -827,15 +693,19 @@ Session::subscribe(const FunctionCallbackInfo<Value>& args, int action)
         RetThrowException(Exception::Error(NEW_STRING(
             "Array of subscription information must be provided.")));
     }
-    if (args.Length() >= 2 && !args[1]->IsUndefined() &&
-        !args[1]->IsNull() && !args[1]->IsObject()) {
+    // Total 2 params. 2nd could be: identity(Object), label(String)
+    if (args.Length() == 2
+        && !(args[1]->IsObject() || args[1]->IsString())) {
         RetThrowException(Exception::Error(NEW_STRING(
-            "Optional identity must be an object.")));
+            "Either identity object or string label "
+            "must be provided as second parameter.")));
     }
-    if (args.Length() >= 3 && !args[2]->IsUndefined() &&
-        !args[2]->IsNull() && !args[2]->IsString()) {
+    // Total 3 params
+    if (args.Length() == 3
+        && !(args[1]->IsObject() && args[2]->IsString())) {
         RetThrowException(Exception::Error(NEW_STRING(
-            "Optional subscription label must be a string.")));
+            "3 parameters combinations: "
+            "Identity object, and label as second & third parameter.")));   
```

Comment:
Just checking -- your error message indicates you should be checking for `args[1]->IsString()`?